tests/test_unittest/test_gtc/test_definitive_assignment_analysis.py

```diff
@@ -36,6 +36,16 @@ def _wrapper(definition):
     return _wrapper
 
 
+@pytest.fixture
+def clear_gtir_caches():
+    from gt4py.backend import module_generator
+    from gtc.passes.gtir_pipeline import GtirPipeline
+
+    GtirPipeline._cache.clear()
+    module_generator._args_data_cache.clear()
+    yield
```

Comment:
Is this here just to be careful?

Reply:
Since the GtirPipeline is now cached per StencilID, not per GtirPipeline instance, only the first occurance of the stencil would print all the warnings / raise exceptions. This fixture resets the cache in between so that the conditions for the tests below are met.